### PR TITLE
Add New Client To Proxy User Request

### PR DESCRIPTION
**Pull request by willmichael**
This is to add a new method that proxies user calls to PDS for the Executer it takes a bearer token and manually constructs the authorization header for the request.

Companion PR to WASH-501 on e3dbsearchservice, waiting on merge of WASH-500 to make PR for that.

### CODE REVIEW

File: pdsClient/pdsClient.go [hunk 1]
```diff
@@ -1,110 +1,110 @@
...
-    if err != nil {
-        return result, err
-    }
-    request, err := http.NewRequest("POST", c.Host+"/"+PDSServiceBasePath+"/search", &buf)
-    if err != nil {
-        return result, err
-    }
-    err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
-    return result, err
+func (c *E3dbPDSClient) ListRecords(ctx context.Context, authToken string, params ListRecordsRequest) (*ListRecordsResult, error) {
```
Comment (galxy25) [inline pdsClient/pdsClient.go]: This should be a separate method, we want to have one method that "normal" clients can use and one for proxy clients(like the executor) to use.

File: main.go [hunk 2]
```diff
@@ -1,54 +1,75 @@
...
-        requestURL := request.URL.String()
-        return &HTTPError{
-            StatusCode: response.StatusCode,
-            URL:        requestURL,
-            message:    fmt.Sprintf("e3db: %s: server http error %d", requestURL, response.StatusCode),
-        }
-    }
-    err = json.NewDecoder(response.Body).Decode(&result)
-    return err
+	client := httpAuthorizer.AuthHTTPClient(ctx)
```
Comment (lkwdwrd) [inline main.go]: So this and the next function look very nearly identical. Is there a reason not to combine the common code?

File: pdsClient/pdsClient.go (hunk 1 above)
Comment (galxy25) [reply pdsClient/pdsClient.go]: Do the tests still pass with this update as is?

Comment (willmichael) [reply pdsClient/pdsClient.go]: It does not (my mistake I did not commit updated test), but I have separated the methods and added line to test both methods.

Comment (galxy25) [reply pdsClient/pdsClient.go]: Awesome!

File: main.go (hunk 2 above)
Comment (willmichael) [reply main.go]: They have distinctly different purposes so I like having them in different functions, but I could probably make a helper method to take out some of the repeated code as I just did with other methods in pdsClient. In this case I'm not sure if I see value, but maybe you see value in refactoring the two methods into one//adding a helper @galxy25?

File: pdsClient/pdsClient.go [hunk 3]
```diff
@@ -1,110 +1,113 @@
...
+	if err != nil {
+		return result, err
+	}
+	err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
+	return result, err
+}
+
+// createListRecordsRequest isolates duplicate code in creating http search request.
+func createRequest(method string, path string, params interface{}) (*http.Request, error) {
+	var buf bytes.Buffer
```
Comment (willmichael) [inline pdsClient/pdsClient.go]: Do you think this refactor obfuscates the functions it takes code out of? @galxy25

File: main.go (hunk 2 above)
Comment (galxy25) [reply main.go]: Definitely two methods as they will be used for distinctly  different purposes, but there would be value in extracting the common code into a helper.

Comment (lkwdwrd) [reply main.go]: Sure -- helper seems like the right call here. The signatures of these functions are different, it just seems like this request pattern works exactly the same when it is used so may be worth abstracting. 👍

**Pull request merged**